backend/src/Designer/Services/Implementation/TextsService.cs

```diff
@@ -325,20 +345,89 @@ private async Task UpdateKeysInLayoutsInLayoutSet(string org, string app, string
+                {
+                    await altinnAppGitRepository.SaveLayout(layoutSetName, layoutName, layout);
+                }
+            }
+        }
+
+        private static void UpdateKeyInLayoutObject(JsonNode layoutObject, TextIdMutation mutation)
+        {
+            if (layoutObject["textResourceBindings"] is JsonObject)
+            {
+                layoutObject["textResourceBindings"] = UpdateTextResourceKeys(layoutObject["textResourceBindings"], mutation);
+            }
+
+            if (layoutObject["options"] is JsonArray optionsArray)
+            {
+                var options = optionsArray.Deserialize<List<Option>>();
+                if (options != null)
+                {
+                    UpdateOptionListKeys(options, mutation);
+                    layoutObject["options"] = JsonSerializer.SerializeToNode(options);
+                }
+            }
+
+            if (layoutObject["source"] is JsonObject)
+            {
+                UpdateSourceKeys(layoutObject["source"], mutation);
+            }
+        }
+
+        private static void UpdateSourceKeys(JsonNode source, TextIdMutation mutation)
+        {
+            JsonElement jsonElement = source["label"].AsValue().GetValue<JsonElement>();
+            if (jsonElement.ValueKind == JsonValueKind.String && jsonElement.GetString() == mutation.OldId && mutation.NewId.HasValue)
+            {
+                source["label"] = mutation.NewId.Value;
+            }
+        }
+
+        private async Task UpdateKeysInOptionLists(string org, string app, string developer, List<TextIdMutation> keyMutations)
+        {
+            string[] optionListIds = _optionsService.GetOptionsListIds(org, app, developer);
+            foreach (string optionListId in optionListIds)
+            {
+                List<Option> options = await _optionsService.GetOptionsList(org, app, developer, optionListId);
+                foreach (TextIdMutation mutation in keyMutations)
+                {
+                    UpdateOptionListKeys(options, mutation);
+                }
+                await _optionsService.CreateOrOverwriteOptionsList(org, app, developer, optionListId, options);
+            }
+        }
+
+        private static void UpdateOptionListKeys(List<Option> options, TextIdMutation keyMutation)
+        {
+            foreach (Option option in options)
+            {
+                if (option.Label == keyMutation.OldId && keyMutation.NewId.HasValue)
+                {
+                    option.Label = keyMutation.NewId.Value;
+                }
+                if (option.Description == keyMutation.OldId && keyMutation.NewId.HasValue)
+                {
+                    option.Description = keyMutation.NewId.Value;
+                }
+                if (option.HelpText == keyMutation.OldId && keyMutation.NewId.HasValue)
+                {
+                    option.HelpText = keyMutation.NewId.Value;
+                }
             }
         }
```

Comment:
I suggest checking the `keyMutation.NewId.HasValue` before entering the foreach loop, something like:

```suggestion
        private static void UpdateOptionListKeys(List<Option> options, TextIdMutation keyMutation)
        {
            if (!keyMutation.NewId.HasValue)
            {
                return;
            }

            foreach (Option option in options)
            {
                if (option.Label == keyMutation.OldId)
                {
                    option.Label = keyMutation.NewId.Value;
                }
                if (option.Description == keyMutation.OldId)
                {
                    option.Description = keyMutation.NewId.Value;
                }
                if (option.HelpText == keyMutation.OldId)
                {
                    option.HelpText = keyMutation.NewId.Value;
                }
            }
        }
```